fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/expressions/Comparisons.java

```diff
@@ -1349,22 +1348,25 @@ public ListComparison(@Nonnull Type type, @Nonnull List comparand) {
                 case IN:
                     break;
                 default:
-                    throw new RecordCoreException("ListComparison only supports EQUALS, NOT_EQUALS and STARTS_WITH");
+                    throw new RecordCoreException("ListComparison only supports EQUALS, NOT_EQUALS, STARTS_WITH and IN");
             }
-            if (comparand == null || (this.type == Type.IN && comparand.stream().anyMatch(o -> o == null))) {
-                throw new NullPointerException("List comparand is null, or contains null");
+            if (!ignoreNulls && this.type == Type.IN && comparand.stream().anyMatch(Objects::isNull)) {
```

Comment:
I am not sure this is right. If we rewrite into an in join and we encounter a a `NULL` in the in list, that is currently not correctly treated (a non-null return in a method would then return `NULL`).

Reply:
corrected

Reply:
As per our talk, reverted the support for NULLs